src/waveform/renderers/waveformwidgetrenderer.h

```diff
@@ -170,6 +170,13 @@ class WaveformWidgetRenderer {
         return renderer;
     }
 
+    template<class T_Renderer, typename T_Arg>
+    inline T_Renderer* addRenderer(T_Arg arg) {
+        T_Renderer* renderer = new T_Renderer(this, arg);
+        m_rendererStack.push_back(renderer);
+        return renderer;
+    }
+
```

Comment:
whats the difference between this template and the one above? This shouldn't even compile IMO because its ambiguous which should be chose in the one parameter case.

Reply:
You are absolutely right, I somehow missed that the above allowed for arguments.